### PR TITLE
Scheduler e2e tests

### PR DESCRIPTION
**Pull request by cicoyle**
Scheduler e2e test working for http & grpc protocols. Schedule jobs & ensure the apps receive the jobs. The grpc app spins up both a grpc & http server. http for test driver calls, grpc for calls to dapr.

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -291,6 +292,7 @@ docker-deploy-k8s: check-docker-env check-arch
 		--set global.imagePullPolicy=$(PULL_POLICY) --set global.imagePullSecrets=${DAPR_TEST_REGISTRY_SECRET} \
 		--set global.mtls.enabled=${DAPR_MTLS_ENABLED} \
 		--set dapr_placement.cluster.forceInMemoryLog=$(FORCE_INMEM) \
+#		--set dapr_scheduler.replicaCount=$(DAPR_SCHEDULER_REPLICAS) \
```
Comment (JoshVanL) [inline Makefile]: Uncomment

File: pkg/runtime/scheduler/streamer.go [hunk 2]
```diff
@@ -144,14 +145,29 @@ func (s *streamer) invokeApp(ctx context.Context, job *schedulerv1pb.WatchJobsRe
...
 	}
 
-	statusCode := response.Status().GetCode()
-	switch codes.Code(statusCode) {
-	case codes.OK:
-		log.Debugf("Sent job: %s to app", job.GetName())
-	case codes.NotFound:
-		log.Errorf("non-retriable error returned from app while processing triggered job %v: %s. status code returned: %v", job.GetName(), statusCode)
+	switch s.isHTTP {
+	case true:
```
Comment (JoshVanL) [inline pkg/runtime/scheduler/streamer.go]: Please use `if s.isHTTP { ... } else {`. `switch` is a bit awkward for booleans.

File: pkg/runtime/scheduler/streamer.go [hunk 3]
```diff
@@ -144,14 +145,29 @@ func (s *streamer) invokeApp(ctx context.Context, job *schedulerv1pb.WatchJobsRe
...
+		statusCode := int(response.Status().GetCode())
+		switch {
+		case statusCode >= 200 && statusCode <= 299:
+			log.Debugf("Sent job: %s to app", job.GetName())
```
Comment (JoshVanL) [inline pkg/runtime/scheduler/streamer.go]:
```suggestion
			log.Debugf("Sent job %s to app", job.GetName())
```

File: pkg/runtime/scheduler/streamer.go [hunk 4]
```diff
@@ -144,14 +145,29 @@ func (s *streamer) invokeApp(ctx context.Context, job *schedulerv1pb.WatchJobsRe
...
+			return nil
+		default:
+			log.Errorf("unexpected status code returned from app while processing triggered job %s. status code returned: %v", job.GetName(), statusCode)
+		}
 	default:
-		log.Errorf("unexpected status code returned from app while processing triggered job %s. status code returned: %v", job.GetName(), statusCode)
+		statusCode := response.Status().GetCode()
+		switch codes.Code(statusCode) {
+		case codes.OK:
+			log.Debugf("Sent job: %s to app", job.GetName())
```
Comment (JoshVanL) [inline pkg/runtime/scheduler/streamer.go]:
```suggestion
			log.Debugf("Sent job %s to app", job.GetName())
```

File: pkg/runtime/scheduler/streamer.go (hunk 2 above)
Comment (JoshVanL) [reply pkg/runtime/scheduler/streamer.go]: q: why are we making a distinction for app protocol at all?

File: tests/apps/schedulerapp/.cache-include [hunk 5]
```diff
@@ -0,0 +1 @@
+../utils/*.go
```
Comment (JoshVanL) [inline tests/apps/schedulerapp/.cache-include]: nit: end file in newline

File: tests/apps/schedulerapp_grpc/.cache-include [hunk 6]
```diff
@@ -0,0 +1 @@
+../utils/*.go
```
Comment (JoshVanL) [inline tests/apps/schedulerapp_grpc/.cache-include]: nit: end file with newline

File: tests/dapr_tests.mk [hunk 7]
```diff
@@ -258,7 +261,7 @@ create-test-namespace:
 	kubectl create namespace $(DAPR_TEST_NAMESPACE)
 
 delete-test-namespace:
-	kubectl delete namespace $(DAPR_TEST_NAMESPACE) aa
```
Comment (JoshVanL) [inline tests/dapr_tests.mk]: Why? Deleting the namespace `aa` is intentional

**Review by JoshVanL — CHANGES_REQUESTED** [review]

File: pkg/runtime/scheduler/streamer.go (hunk 2 above)
Comment (cicoyle) [reply pkg/runtime/scheduler/streamer.go]: We need the distinction to acct for http 200 throwing the unexpected error. I showed this to you before when you had your refactor PR open. I'll try to see if I have it saved

File: tests/apps/schedulerapp_grpc/.cache-include (hunk 6 above)
Comment (cicoyle) [reply tests/apps/schedulerapp_grpc/.cache-include]: none of the other `.cache-include` files have a newline from what I saw

File: tests/dapr_tests.mk (hunk 7 above)
Comment (cicoyle) [reply tests/dapr_tests.mk]: isn't it: `k delete ns <myns>`? I think aa is a typo and shouldnt be there

File: pkg/runtime/scheduler/streamer.go (hunk 2 above)
Comment (cicoyle) [reply pkg/runtime/scheduler/streamer.go]: I thought it was cleaner with the switch

File: tests/apps/schedulerapp/.cache-include (hunk 5 above)
Comment (cicoyle) [reply tests/apps/schedulerapp/.cache-include]: none of the other .cache-include files have a newline from what I saw

File: tests/apps/schedulerapp_grpc/.cache-include (hunk 6 above)
Comment (JoshVanL) [reply tests/apps/schedulerapp_grpc/.cache-include]: All text files should end in a newline :slightly_smiling_face:

File: tests/dapr_tests.mk (hunk 7 above)
Comment (JoshVanL) [reply tests/dapr_tests.mk]: It is not a typo. It is deleting both the `DAPR_TEST_NAMESPACE` and `aa` namespaces.

File: pkg/runtime/scheduler/streamer.go (hunk 2 above)
Comment (cicoyle) [reply pkg/runtime/scheduler/streamer.go]: this: 
```
for scheduler, to an http app at trigger time Im seeing:
ERRO[0237] unexpected status code returned from app while processing triggered job test1. status code returned: 200` 
```

Comment (cicoyle) [reply pkg/runtime/scheduler/streamer.go]: I did update it to use an if statement instead of the switch per your request

Comment (JoshVanL) [reply pkg/runtime/scheduler/streamer.go]: I don't like the fact that we need to make the distinction here, and makes the point of having an `app channel` abstraction a bit pointless if we always need to do the check anyway. Can we standardize on the error code returned by both protocol channels, converting HTTP status codes to gRPC codes?

Nit on the code in that function- the `if` statement checking for the 200 range should be at the top :sweat_smile: 
https://github.com/dapr/dapr/blob/master/pkg/messaging/v1/util.go#L299

Comment (cicoyle) [reply pkg/runtime/scheduler/streamer.go]: Can we do the conversion in a follow up PR to dapr/dapr? I can add an issue in dapr/dapr for it to not forget and tag it as a good first issue.

Comment (cicoyle) [reply pkg/runtime/scheduler/streamer.go]: I agree on the nit re: the 200 should go at the top

Comment (JoshVanL) [reply pkg/runtime/scheduler/streamer.go]: Please can we add a TODO in the code?

Comment (cicoyle) [reply pkg/runtime/scheduler/streamer.go]: I added the code todo and moved the 200 check up

**Review by JoshVanL — APPROVED** [review]

**Pull request merged**